### PR TITLE
Fix single-item and collection deletion api doc

### LINKED ISSUE #105104: openapi schema indicates incorrect return types for delete and deletecollection operations
1. Many (most?) types do not specify ReturnDeletedObject in their storage configuration
2. For those types, the generated schema specifies `v1.Status` as the return type for deleting a single item and collection of items
3. The actual behavior of deleting a single item :
   * Deleting a job with `propagationPolicy=Orphan`, `propagationPolicy=Foreground`, or  no `propagationPolicy` (which defaults to orphan) returns `Job`
   * Deleting a job with `propagationPolicy=Background` returns `Status`
   * Deleting a validationwebhook with `propagationPolicy=Orphan` or `propagationPolicy=Foreground` return `ValidatingWebhookConfiguration`
   * Deleting a validationwebhook with `propagationPolicy=Background` or no `propagationPolicy` (which defaults to background) returns `Status`
4. The actual behavior of deleting a collection of items:
   * Deleting a collection of jobs returns `JobList` in all cases
   * Deleting a collection of validating webhooks returns `ValidatingWebhookConfigurationList` in all cases

So, for objects which do not set `ReturnDeletedObject: true`, the generated schema is incorrect in some cases for the single-item delete, and incorrect in all cases for the collection delete.

However, if we change to set `ReturnDeletedObject: true` for all objects, we will start returning objects in some cases where we previous returned Status, and will break clients successfully using those deletion paths that could not handle an object return.

Since all clients have to be able to handle Status in all cases (since it can be returned for forbidden errors, not found errors, etc), I think the safest thing to do is to make schema generation stop paying attention to `ReturnDeletedObject` and have it indicate the object can be a potential return value for single-item and collection deletion.

_Originally posted by @liggitt in https://github.com/kubernetes/kubernetes/issues/93477#issuecomment-921949090_

### PR DESCRIPTION
**Pull request by luyou86**
1. What type of PR is this?
/kind bug
/sig api-machinery

2. What this PR does / why we need it:
openapi schema indicates incorrect return types for delete and deletecollection operations

3. Which issue(s) this PR fixes:
Fixes #105104

4. Does this PR introduce a user-facing change?
NONE

### CODE REVIEW

File: api/openapi-spec/swagger.json [hunk 1]
```diff
@@ -51920,7 +51920,7 @@
           "200": {
             "description": "OK",
             "schema": {
-              "$ref": "#/definitions/io.k8s.apimachinery.pkg.apis.meta.v1.Status"
+              "$ref": "#/definitions/io.k8s.apimachinery.pkg.apis.meta.v1.ObjectMeta",
```
Comment (liggitt) [inline api/openapi-spec/swagger.json:51923]: this file is generated... the type registered for Writes/Returns has to be fixed in staging/src/k8s.io/apiserver/pkg/endpoints/installer.go for the DELETE and DELETECOLLECTION cases

then, run `hack/update-openapi-spec.sh` to regenerate this file

also, this isn't just an issue for the jobs API, it affects all delete/deletecollection types

**Comment by liggitt** [comment]
this is actually a sig-apimachinery issue, since the API server generation of the openapi schema is incorrect for all objects

/remove-sig apps
/sig api-machinery

**Comment by apelisse** [comment]
@Jefftree Would you be able to look at this when you have a chance? I know you've been spending too much time looking at this code recently ;-)

**Comment by Jefftree** [comment]
Might need to run `hack/update-openapi-spec.sh` to regenerate the openapi again.  OTOH, this is something that will be fixed by OpenAPI v3, specifically the `oneOf` field, and this is planned for alpha this release (1.23). For v2, it seems like regardless of what we choose to return (object vs status), it will always be wrong for a portion of the cases as liggitt described.

**Comment by luyou86** [comment]
> Might need to run `hack/update-openapi-spec.sh` to regenerate the openapi again. OTOH, this is something that will be fixed by OpenAPI v3, specifically the `oneOf` field, and this is planned for alpha this release (1.23). For v2, it seems like regardless of what we choose to return (object vs status), it will always be wrong for a portion of the cases as liggitt described.

Thank for your help. it is true that "oneOf" only is supported in OpenAPI v3. in staging/src/k8s.io/apisever/pkg/endpoints/installer.go, i remove deleteReturnsDeletedObject so that the Writes and Returns function of "DELETE" and "DELETECOLLECTION" always output "versionedStatus" variable. 

not sure it is ok

**Comment by Jefftree** [comment]
Per @liggitt's comment https://github.com/kubernetes/kubernetes/issues/105104#issuecomment-923047118,

> I don't see a way to make the schema correct, since a 200 can return either a Status or Job, depending on delete options specified... if we had to pick just one, I think I'd pick the specific object type, not status, since clients have to be generally prepared to receive a status object from any operation on any endpoint

Seems to make more sense to return the object instead of the status? Nevertheless this is still a breaking change on the client, and with OpenAPI v3 in the works (for `oneOf`) I feel that we should use that to solve this problem completely rather than adding another bandaid solution.

**Comment by luyou86** [comment]
why swagger.json conflict after i push regenerate swagger doc

**Pull request closed without merging**